### PR TITLE
[5.2] add dynamic orwhere and fix a test that do nothing

### PR DESCRIPTION
**Pull request by wanghanlin**
#12795

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1090,6 +1090,22 @@ public function dynamicWhere($method, $parameters)
     }
 
     /**
+     * Handles dynamic "orWhere" clauses to the query.
+     *
+     * @param  string  $method
+     * @param  string  $parameters
+     * @return $this
+     */
+    public function dynamicOrWhere($method, $parameters)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: This surely doesn't need to be public?

Comment (shwhl) [inline src/Illuminate/Database/Query/Builder.php]: function dynamicWhere is public, so I think it's ok to be public?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: It shouldn't be public.

Comment (shwhl) [inline src/Illuminate/Database/Query/Builder.php]: Sure, should I have to change `dynamicWhere` to protected as well?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: No. That'll need doing in 5.3.

Comment (shwhl) [inline src/Illuminate/Database/Query/Builder.php]: Ok, done.

**Comment by taylorotwell** [comment]
Holding off on this.

**Pull request closed without merging**